pilot/pkg/xds/endpoint_builder.go

```diff
@@ -279,8 +281,11 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 		if isClusterLocal && (clusterID != b.clusterID) {
 			continue
 		}
-
 		for _, ep := range endpoints {
+			// TODO(nmittler): Consider merging discoverability policy with cluster-local
+			if !b.push.IsDiscoverableFromProxy(b.service, ep, b.proxy) {
```

Comment:
way outside of scope - I wonder if it would be useful to have a new debug endpoint like endpointz, but it shows all endpoints and for any that are filtered explain why (different network, not exported service, etc)

Reply:
Yeah good idea. Would be a good follow-on PR.